src/core/ext/transport/chttp2/transport/hpack_parser.h

```diff
@@ -27,84 +27,205 @@
 #include "src/core/ext/transport/chttp2/transport/hpack_table.h"
 #include "src/core/lib/transport/metadata.h"
 
-typedef struct grpc_chttp2_hpack_parser grpc_chttp2_hpack_parser;
+namespace grpc_core {
 
-typedef grpc_error_handle (*grpc_chttp2_hpack_parser_state)(
-    grpc_chttp2_hpack_parser* p, const uint8_t* beg, const uint8_t* end);
+class HPackParser {
+ public:
+  enum class Boundary { None, EndOfHeaders, EndOfStream };
+  enum class Priority { None, Included };
 
-struct grpc_chttp2_hpack_parser_string {
-  bool copied;
-  struct {
-    grpc_slice referenced;
+  // User specified structure called for each received header.
+  using Sink = std::function<grpc_error_handle(grpc_mdelem)>;
```

Comment:
Are there any alternative names to `Sink`?

Reply:
I mean there's a universe of names that are not `Sink`. Can you tell me what your objection is though, in order to shrink that universe?

Reply:
Ignore the comment. I just did a quick search on the usage, and it seems to follow that convention here. (I haven't used it myself yet so didn't know what it meant. I'd probably have used something like OnHeaderCallback/Action)